pkg/k8s/watchers/pod.go

```diff
@@ -461,7 +461,8 @@ func (k *K8sWatcher) deleteK8sPodV1(pod *slim_corev1.Pod) error {
 
 func (k *K8sWatcher) genServiceMappings(pod *slim_corev1.Pod, podIPs []string, logger *logrus.Entry) []loadbalancer.SVC {
 	var svcs []loadbalancer.SVC
-	for _, c := range pod.Spec.Containers {
+	containers := append(pod.Spec.InitContainers, pod.Spec.Containers...)
+	for _, c := range containers {
```

Comment:
Actually let's do it like this, just to be safe:
```golang
nContainers := len(pod.Spec.InitContainers) + len(pod.Spec.Containers)
var c slim_corev1.Container
for i := 0; i < nContainers; i++ {
   if i<len(pod.Spec.InitContainers){
     c = pod.Spec.InitContainers[i]
   }else{
    c =pod.Spec.Containers[i-len(pod.Spec.InitContainers)]
   }
....
```

Reply:
https://go.dev/play/p/_C5VYvaJ0i1

Reply:
:exploding_head: 

What about the following?
```go
var containers []Container
containers = append(containers, pod.Spec.InitContainers...)
containers = append(containers, pod.Spec.Containers...)
for _, c := range containers {
```
Seems easier to parse.

Reply:
yep, that also works although we are kind of unnecessarily allocating memory